add-ons/updating/tb78/convert-legacy.md

```diff
@@ -0,0 +1,146 @@
+
+Full examples for [a simple function with parent and child implementations](https://firefox-source-docs.mozilla.org/toolkit/components/extensions/webextensions/functions.html) and [events add-ons can listen for](https://firefox-source-docs.mozilla.org/toolkit/components/extensions/webextensions/events.html) are available in the Firefox source documentation.
+
+If your experiment is so complex that it does not reasonably fit into a single source file, you can use JavaScript modules just like in legacy extensions with some additional boilerplate:
+```
+const { ExtensionParent } = ChromeUtils.import(
+    "resource://gre/modules/ExtensionParent.jsm");
+const extension = ExtensionParent.GlobalManager.getExtension(
+    "insert-your-extension-id-here@example.com");
+const { /* ... exported symbols ... */ } =
+    ChromeUtils.import(extension.getURL("path/to/module.jsm"));
+
+// when unloading: (safe to call even if the import is conditional / elsewhere!)
+Components.utils.unload(extension.getURL("path/to/module.jsm"));
+```
+
+As experiments usually run in the main process and have unrestricted access to any aspect of Thunderbird, they are expected to require updates for each new version of Thunderbird. To reduce the maintainance burden in the future, it is in your own interest to use experiments only to the extent necessary for the add-on.
+
+Best practice: try to write APIs that would be useful for a wide range of add-ons, not just the one you're porting. That way, you can later on propose the API you designed for inclusion in Thunderbird, with your add-on serving as reference implementation. If your APIs become a part of Thunderbird, you no longer need to maintain them as part of the add-on.
+
+
+## Replacing options
+
+Option windows are replaced by [settings pages](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Implement_a_settings_page) defined in `manifest.json`:
+```
+"options_ui": {
+  "page": "options.html"
+},
+```
+Instead of an XUL dialog, use the specified HTML document will be accessible to the user through the Add-on manager. From that document, all WebExtension and MailExtension APIs can be accessed in the same way as form a background script. The settings themselves should be stored using one of the new APIs to store data, such as [`storage`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage), so it may be necessary to add an experiment to migrate existing settings from nsIPrefBranch or other mechanisms not accessible through modern APIs.
+
```

Comment:
I would like to link to and explain: https://github.com/jobisoft/quicktext/blob/pro/manifest.json#L24-L30
which is the described migration API.

I also would move this section way up, because once you migrated your options, there are no longer in the legacy pref system (at least should not) and than you have to go back and adjust your other code to use the storage.

Reply:
Feel free to do so!